### ISSUE TITLE
[VisibilityDetector] VisibilityDetector callback doesn't get triggered in TalkBack mode

### ISSUE THREAD

**Issue by SSK0525** [google/flutter.widgets#558]
## Problem description

The onVisibilityChanged callback of VisibilityDetector is triggered only once in TalkBack mode. In our case, we have a widget wrapped with VisibilityDetector placed at the bottom of the page. When the user scrolls and the widget becomes visible, we perform an action based on its visibility. However, in TalkBack mode, we’ve observed that the callback is triggered only once.

## Steps to reproduce

1. Place a VisibilityDetector widget and enable TalkBack A11Y mode.
2. Use volume up/down key to scroll the page to check the VisibilityDetector callback triggers.

## Expected behavior

In TalkBack mode, the VisibilityDetector callback should be triggered whenever the widget becomes visible or invisible.

## Actual behavior

The VisibilityDetector callback is triggered only once in TalkBack mode.

## Environment

Android OS 15
